avogadro/qtplugins/closecontacts/closecontacts.cpp

```diff
@@ -102,6 +104,91 @@ void CloseContacts::process(const Molecule &molecule, Rendering::GroupNode &node
         lines->addDashedLine(pos.cast<float>(), npos.cast<float>(), color, 8);
     }
   }
+
+  // Add charged atoms
+  Array<Vector3> positions;
+  Array<signed char> charges;
+  Array<Index> residues;
+  for (Index i = 0; i < molecule.atomCount(); ++i) {
+    if (molecule.formalCharge(i) != 0) {
+      if (!isAtomEnabled[i])
+        continue;
+      positions.push_back(molecule.atomPosition3d(i));
+      charges.push_back(molecule.formalCharge(i));
+      residues.push_back(Index(0) - 1);
+    }
+  }
+
+  // Add predicted charged atoms from residues
+  for (const auto &r: molecule.residues()) {
+    for (const auto &a: r.residueAtoms()) {
+      if (molecule.formalCharge(a.index()) != 0)
+        continue;
+      bool gammaOxygen = false;
+      bool doubleBond = false;
+      bool alphaDoubleBond = false;
+      auto bonds = molecule.getAtomBonds(a.index());
+      auto orders = molecule.getAtomOrders(a.index());
+      for (Index i = 0; i < bonds.size(); i++) {
```

Comment:
Please add some comments about what you're doing here. You're trying to find particular atoms, right?

Do you want to look at residue atom names? `Residue::atomNameMap()` for example.

Check out LigandExpo:
- http://ligand-expo.rcsb.org/ld-search.html
- e.g. http://ligand-expo.rcsb.org/reports/H/HIS/HIS_D3L3.gif

Now some files will come in with problematic labels, but definitely check atom names first, because they'll save a lot of time.